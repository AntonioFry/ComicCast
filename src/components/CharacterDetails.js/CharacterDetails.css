.character-details-section {
  display: flex;
  padding: 100px
}

.general-info-article {
  background-color: white;
  border: 1px solid black;
  width: 200px
}

.character-details-img {
  border-bottom: 1px solid black; 
  width: 200px;
  height: 250px;
}

.details-name,
.details-full-name {
  margin: 15px 5px;
}

.details-right-section {
  margin: 0px 0px 0px 20px;
}

.trivia-article,
.overview-article {
  background-color: white;
  border: 1px solid black;
}

.character-publisher,
.details-first-appearance,
.details-overview {
  line-height: 1.5em;
  margin: 10px 5px;
}

.trivia-header {
  margin: 20px 0px 10px;
}