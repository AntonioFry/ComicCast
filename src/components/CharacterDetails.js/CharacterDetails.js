import React from 'react';
import './CharacterDetails.css';

export const CharacterDetails = (props) => {
  const { image, overview, name,  full_name, first_appearance, publisher } = props;
  return (
    <section className="character-details-section">
      <article className="general-info-article">
        <img className="character-details-img" src={image} alt={`the character ${name}`}/>
        <div className="general-info-text-container">
          <p className="details-name">Name: {name}</p>
          <p className="details-full-name">Full Name: {full_name}</p>
        </div>
      </article>
      <section className="details-right-section">
        <h2 className="overview-header" >Overview</h2>
        <article className="overview-article">
          <p className="details-overview">Overview: {overview}</p>
        </article>
        <h2 className="trivia-header" >Trivia</h2>
        <article className="trivia-article">
          <p className="details-first-appearance">First Appearance: {first_appearance}</p>
          <p className="character-publisher">Publisher: {publisher}</p>
        </article>
      </section>
    </section>
  )
}

export default CharacterDetails;