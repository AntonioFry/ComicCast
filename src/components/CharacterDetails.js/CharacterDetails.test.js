import React from 'react';
import { CharacterDetails } from './CharacterDetails';
import { shallow } from 'enzyme';

describe('CharacterDetails', () => {
  let wrapper;
  let props;

  beforeEach(() => {
    props = {
      image: '',
      overview: '',
      name: '',
      full_name: '',
      first_appearance: '',
      publisher: ''
    }
    wrapper = shallow(<CharacterDetails
      id={1}
      key={1}
      image={props.image}
      overview={props.overview}
      publisher={props.name}
      name={props.name}
      full_name={props.full_name}
      first_appearance={props.first_appearance}
      />)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});