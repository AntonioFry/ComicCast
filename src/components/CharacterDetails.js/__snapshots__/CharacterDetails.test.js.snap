// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterDetails should match the snapshot 1`] = `
<section
  className="character-details-section"
>
  <article
    className="general-info-article"
  >
    <img
      alt="the character "
      className="character-details-img"
      src=""
    />
    <div
      className="general-info-text-container"
    >
      <p
        className="details-name"
      >
        Name: 
      </p>
      <p
        className="details-full-name"
      >
        Full Name: 
      </p>
    </div>
  </article>
  <section
    className="details-right-section"
  >
    <h2
      className="overview-header"
    >
      Overview
    </h2>
    <article
      className="overview-article"
    >
      <p
        className="details-overview"
      >
        Overview: 
      </p>
    </article>
    <h2
      className="trivia-header"
    >
      Trivia
    </h2>
    <article
      className="trivia-article"
    >
      <p
        className="details-first-appearance"
      >
        First Appearance: 
      </p>
      <p
        className="character-publisher"
      >
        Publisher: 
      </p>
    </article>
  </section>
</section>
`;
